app/service_terms.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+)
+
+func (a *App) CreateServiceTerms(text, userId string) (*model.ServiceTerms, *model.AppError) {
+	serviceTerms := &model.ServiceTerms{
+		Text:   text,
+		UserId: userId,
+	}
+
+	if _, err := a.GetUser(userId); err != nil {
+		return nil, err
+	}
+
+	if result := <-a.Srv.Store.ServiceTerms().Save(serviceTerms); result.Err != nil {
```

Comment:
We've been changing our error handling strategy to be more idiomatic (i.e. early exits without redundant else conditions). Can you tweak this block to look like the following instead?
```go
result := <-a.Srv.Store.ServiceTerms().Save(...);
if result.Err != nil {
    return nil, result.Err
}

return result.Data.(*model.ServiceTerms), nil
```

Reply:
Done.